### ISSUE TITLE
Ruby: normalize extra parameters parenthesis

### ISSUE THREAD

**Issue by becojo** [semgrep/semgrep#5222]
**Describe the bug**
In Ruby, given the semgrep pattern `method("...")`, semgrep will match `method("foo")` but not the equivalent `method ("foo")`

You could extend the rule to account for that specific case, but I think semgrep should normalize the code to not consider redundant characters. The pattern `method("...")` should match `method "..."`, `method("...")`, `method ("...")`, `method (("..."))`, `method(("..."))`, ... as they are all equivalent method calls with a single string argument.

**To Reproduce**

https://semgrep.dev/s/x9Oy

```
rules:
- id: ruby-method-call
  pattern: method("...")
  message: Semgrep found a match
  languages: [ruby]
  severity: WARNING
```

Test cases:
```
# ruleid: ruby-method-call
method("adsf") # pass

# ruleid: ruby-method-call
method "adsf" # pass

# ruleid: ruby-method-call
method ("adsf") # fail

# ruleid: ruby-method-call
method (("adsf")) # fail

# ruleid: ruby-method-call
method(("adsf")) # fail
```

**Expected behavior**
All test cases pass

**Screenshots**

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [X] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
If not using semgrep.dev: are you running off docker, an official binary, a local build?

**Use case**
- Prevent false negatives. Semgrep should be able to find the targeted code even if the code is formatted poorly.

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1294/ruby-normalize-extra-parameters-parenthesis. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by ajbt200128** [comment]
@Becojo So it turns out this issue was really interesting, but TLDR we've decided there's too much nuance in how ruby interpreters work for us to support this. The long explanation is as follows:

Ruby allows the omitting of parentheses around method arguments such as in the example `method "adsf"`. So any ruby parser will parse that the same as `method("asdf")`, and clearly they are semantically the same too, and this is in the ruby standards. This is where it gets really tricky though, as ruby will allow you to surround any expression with parentheses (for the purposes of precedence, like in most languages (so `(2*2)-2 != 2*(2-2)`). This means that if we want, we could have `method("asdf")` OR `method(("asdf"))`. The thing here is though, is that by doing this we are explicitly stating "we want to ensure precedence for the expression inside the parentheses". 

What we ended up finding is that _some_ ruby parsers would parse `method("asdf")` and `method(("asdf"))` the same, since at least in this case, where the argument is a literal value, they are semantically the same, but some parsers do not, as they choose to keep the fact that we have that operator precedence. Interestingly then, this implies that `method("asdf")` and `method "asdf"` are syntactically + semantically the same, and so are `method(("asdf"))` and `method ("asdf")` as the former of each pair has parentheses surrounding the argument list, and the latter does not. 

The big thing here though is then that since `method("asdf")` is not always parsed the same as `method(("asdf"))`, as they are not syntactically the same, but are semantically the same, we realized there can be some edge cases where the argument inside the operator precedence parentheses is not a literal, and it may not semantically be the same. 

Since ruby parsers and the actual standards aren't decided on this, we won't make the decision either, as to implement this functionality, we'd have to modify the ruby parser we use which is maintained by the community, is used by not just us, and it is not our place to decide how ruby should work. Here's a good [stack overflow question](https://stackoverflow.com/questions/17036166/why-do-i-need-extra-parentheses-for-conditional-modifier) on this.

Bonus: Syntactically `method (("asdf"))` is the same as `method((("asdf")))`, but not the same as `method ("asdf")` nor `method "asdf"`, though in this case they are all semantically the same (kind of).
